message/subscriber.go

```diff
@@ -15,3 +15,14 @@ type Subscriber interface {
 	// Close closes all subscriptions with their output channels and flush offsets etc. when needed.
 	Close() error
 }
+
+type SubscribeInitializer interface {
+	// SubscribeInitialize can be called to initialize subscribe before consume.
+	// When calling Subscribe before Publish, SubscribeInitialize should be not required.
+	//
+	// Not every Pub/Sub requires this initialize and it may be optional for performance improvements etc.
+	// For detailed SubscribeInitialize functionality, please check Pub/Subs godoc.
+	//
+	// Implementing SubscribeInitialize is not obligatory.
+	SubscribeInitialize(topic string) error
```

Comment:
with `Initialize` you don't know that it is subscribe or publish initialize, and `CreateSubscription` is too narrow (it may be good for GC PubSub, but for rabbit it doesn't make sense)